stable/app/templates/cron.yaml

```
@@ -46,11 +47,13 @@ spec:
               {{- end }}
               args:
               {{- toYaml $job.args | nindent 16 }}
+              {{- if $secretConfig }}
               envFrom:
                 - configMapRef:
                     name: {{ $fullName }}
                 - secretRef:
                     name: {{ $fullName }}-secret
+              {{- end }}
```

let's use the value directly and move the if condition only for the secret and let the configmap as is
```suggestion
              envFrom:
                - configMapRef:
                    name: {{ $fullName }}
              {{- if .Values.secretConfig }}
                - secretRef:
                    name: {{ $fullName }}-secret
              {{- end }}
```

---

using this directly is giving issue, as this in range global variable should be used as mentioned here as well
https://stackoverflow.com/questions/57475521/ingress-yaml-template-returns-error-in-renderring-nil-pointer-evaluating-int#:~:text=What%20you%27re%20seeing,Values%20as%20expected.

---

this somehow works now, if we put the condition just for secretRef
resolved @haveiss